core/src/main/scala/org/apache/spark/deploy/SparkPipelines.scala

```diff
@@ -65,10 +65,10 @@ object SparkPipelines extends Logging {
         } else if (opt == "--class") {
           logError("--class argument not supported.")
           throw SparkUserAppException(SparkExitCode.EXIT_FAILURE)
-        } else if (opt == "--conf" && value.startsWith("spark.api.mode=")) {
-          if (value.toLowerCase(Locale.ROOT) != "spark.api.mode=connect") {
+        } else if ((opt == "--conf" || opt == "-c") && value.startsWith("spark.api.mode=")) {
+          if (value.stripPrefix("spark.api.mode=").trim.toLowerCase(Locale.ROOT) != "connect") {
```

Comment:
Use `SparkLauncher.SPARK_API_MODE` or `config.SPARK_API_MODE.key` 🙏

Reply:
updated to use `SparkLauncher.SPARK_API_MODE`